org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLRename.java

```diff
@@ -54,126 +57,171 @@ public XMLRename(XMLExtensionsRegistry extensionsRegistry) {
 	}
 
 	public WorkspaceEdit doRename(DOMDocument xmlDocument, Position position, String newText) {
-		List<TextEdit> textEdits = getRenameTextEdits(xmlDocument, position, newText);
-		Map<String, List<TextEdit>> changes = new HashMap<>();
-		changes.put(xmlDocument.getDocumentURI(), textEdits);
-		return new WorkspaceEdit(changes);
+
+		DOMNode node = getNodeToRename(xmlDocument, position);
+		if (node == null || !node.isElement() || ((DOMElement) node).getTagName() == null) {
+			return createWorkspaceEdit(xmlDocument.getDocumentURI(), Collections.emptyList());
+		}
+
+		List<TextEdit> textEdits = new ArrayList<>();
+		DOMElement element = (DOMElement) node;
```

Comment:
Remove getNodeTorename and call

```java
RenameRequest renameRequest = null;
try {
	renameRequest  = new RenameRequest(xmlDocument, position, newText, extensionsRegistry);
} catch (BadLocationException e) {
	LOGGER.log(Level.SEVERE, "Failed creating RenameRequest", e);
	return null;
}
DOMNode node = renameRequest.getNode(); 

```

Please note that node is element or attribute.